### ISSUE TITLE
ChannelTabControl not handling channel closing correctly

### ISSUE THREAD

**Issue by TheWildTree** [ppy/osu#7465]
Right now after closing a currently active channel, [ChannelTabControl](https://github.com/ppy/osu/blob/master/osu.Game/Overlays/Chat/Tabs/ChannelTabControl.cs) always switches to the channel on its left (the **previous** one). However, as indicated by the comments in the code, the desired behavior would be for it to always switch to the **next** one (on its right side), only exception being if it was the last channel on the list.

This is also mentioned by ppy/osu-framework#3152, though as I'm not experienced enough to investigate this issue framework-wise, I wanted to focus on this specific scenario here.

This problem seems to be caused by conflicting `RemoveChannel` and `tabCloseRequested` functions. 

`tabCloseRequested` seems to launch first after closing a channel, switching the tab if the closed one was currently active. However, `TabContainer.IndexOf(tab)` returns the index counting from the right side of the channel list, making the tab selecting part underneath work incorrectly (it assumes counting from the left):

```csharp
if (tab == SelectedTab && totalTabs > 1)
     // Select the tab after tab-to-be-removed's index, or the tab before if current == last
     SelectTab(TabContainer[currentIndex == totalTabs ? currentIndex - 1 : currentIndex + 1]);
``` 

After this, `RemoveChannel`'s `Current.Value == channel` check is never gonna be true due to `tabCloseRequested` changing the tab beforehand. Even if it did execute, it seems to only set the `Current` value instead of using `SelectTab()`, so it seems kinda obsolete.

I can fix this, but I'm not sure which of these functions is desired to handle the switching and if the order of items being reversed (as indicated by the `IndexOf` call) is intended.

**Comment by iiSaLMaN** [comment]
You're probably looking for [`TabContainer.TabItems`](https://github.com/ppy/osu-framework/blob/7231a22018c314378e4308d7798545b4093f006e/osu.Framework/Graphics/UserInterface/TabControl.cs#L385) where the tab items should be ordered exactly as you see them in screen. (from the left)

**Comment by TheWildTree** [comment]
The problem is that [`TabContainer.TabItems`](https://github.com/ppy/osu-framework/blob/7231a22018c314378e4308d7798545b4093f006e/osu.Framework/Graphics/UserInterface/TabControl.cs#L385) doesn't expose a `IndexOf`, so to get the current index I'd have to loop through the items until I find an active one? If that's the case, `TabContainer.Count - TabContainer.IndexOf(tab)` seems like a better solution for me, if the reversed order is indeed intended.

**Comment by iiSaLMaN** [comment]
That's a very dangerous and bug-prone way to take, i'm not sure what caused the `TabContainer`'s children to be added in an exact reversed order but you're simply reading the `TabContainer`'s items in an unsorted way (from oldest added to newest, and one of the items could possibly been removed).

Just do `var itemsList = TabContainer.TabItems.ToList();` if necessary.

**Comment by TheWildTree** [comment]
Thanks, that makes a lot more sense. I'll make a PR in a minute.

**Comment by TheWildTree** [comment]
Closed by #8270.
